internal/cloudinfo/machine_types.go

```diff
@@ -94,39 +96,42 @@ func (im *InstanceTypeMap) setMachines(cloud string, service string, region stri
 var instanceTypeMap = NewInstanceTypeMap()
 
 func fetchMachineTypes(logger logrus.FieldLogger, cloud string, service string, region string) error {
-	cloudInfoEndPoint := viper.GetString(config.CloudInfoEndPoint)
-	if len(cloudInfoEndPoint) == 0 {
-		return emperror.With(errors.New("missing config"), "propertyName", config.CloudInfoEndPoint)
-	}
-	cloudInfoUrl := fmt.Sprintf(
-		"%s/providers/%s/services/%s/regions/%s/products",
-		cloudInfoEndPoint, cloud, service, region)
-	ciRequest, err := http.NewRequest(http.MethodGet, cloudInfoUrl, nil)
-	if err != nil {
-		return errors.New("error fetching machine types from CloudInfo")
+	ciHostName := viper.GetString(config.CloudInfoHostName)
+	if len(ciHostName) == 0 {
+		return emperror.With(errors.New("missing config"), "ciHostName", config.CloudInfoHostName)
 	}
 
-	ciRequest.Header.Set("Content-Type", "application/json")
-	httpClient := &http.Client{}
+	ciBasePath := viper.GetString(config.CloudInfoBasePath)
+	if len(ciBasePath) == 0 {
+		return emperror.With(errors.New("missing config"), "ciBasePath", config.CloudInfoBasePath)
+	}
 
-	log := logger.WithFields(logrus.Fields{"cloudInfoEndPoint": cloudInfoEndPoint, "cloud": cloud, "region": region, "service": service})
+	log := logger.WithFields(logrus.Fields{"cloudInfoHost": ciHostName, "cloud": cloud, "region": region, "service": service})
 	log.Info("fetching machine types from CloudInfo")
 
-	ciResponse, err := httpClient.Do(ciRequest)
+	transport := httptransport.New(ciHostName, ciBasePath, []string{"https"})
+	cloudInfoClient := client.New(transport, strfmt.Default)
+
+	params := &products.GetProductsParams{
+		Provider: cloud,
+		Region:   region,
+		Service:  service,
+	}
+	getProductsOK, err := cloudInfoClient.Products.GetProducts(params.WithTimeout(60 * time.Second))
 	if err != nil {
 		return emperror.Wrap(err, "error fetching machine types from CloudInfo")
 	}
-	respBody, _ := ioutil.ReadAll(ciResponse.Body)
-	var vmDetails CloudInfoResponse
-	json.Unmarshal(respBody, &vmDetails)
-
-	instanceTypeMap.setMachines(cloud, service, region, vmDetails.Products)
+	if getProductsOK.Payload != nil {
+		instanceTypeMap.setMachines(cloud, service, region, getProductsOK.Payload.Products)
+	} else {
+		return errors.New("error fetching machine types from CloudInfo: empty Payload in response")
```

Comment:
use emperror wrap

Reply:
the error is wrapped in calling function